src/main/java/Duke.java

```diff
@@ -1,10 +1,75 @@
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
+
+    private static boolean isNonAddTaskInput (String input, List<Task> tasks) {
+        if (input.contains("done")) {
+            int taskIndex = Integer.parseInt(input.replaceAll("[^0-9]", ""));
+            tasks.get(taskIndex - 1).markAsDone();
+            Task.numberOfTasks -= 1;
+            return true;
+        }
+
+        else if (input.equals("list")) {
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.print(i + 1 + ".");
+                tasks.get(i).describe();
+            }
+            return true;
+        }
+
+        return false;
+    }
+
+    private static Task addTask (String input, List<Task> tasks) {
+        String taskType = input.split(" ")[0];
+        String descriptionAndTime = input.replaceAll(taskType, "");
+
+        switch (taskType) {
+        case "todo" :
+            Task todoTask = new Todo(input.replaceAll(taskType, ""));
+            tasks.add(todoTask);
+            return todoTask;
+
+        case "deadline" :
+            Task deadlineTask = new Deadline (descriptionAndTime.split("/")[0], descriptionAndTime.split("/")[1]);
```

Comment:
This expression may be a bit long, perhaps you can consider making a new variable for descriptionAndTime.split("/")